tests/unit/workflows/test_workflows_tasks_classifier.py

```diff
@@ -51,6 +57,182 @@
 '''
 
 
+def test_get_classifier_api_url_from_configuration():
+    config = {'CLASSIFIER_API_URL': 'https://inspire-classifier.web.cern.ch/api'}
+
+    with patch.dict(current_app.config, config):
+        expected = 'https://inspire-classifier.web.cern.ch/api/predict/coreness'
+        result = get_classifier_url()
+
+        assert expected == result
+
+
+def test_get_classifier_api_url_returns_none_when_not_in_configuration():
+    config = {'CLASSIFIER_API_URL': ''}
+
+    with patch.dict(current_app.config, config):
+        assert get_classifier_url() is None
+
+
+def test_prepare_payload():
+    record = {
+        'titles': [
+            {
+                'title': 'Effects of top compositeness',
+            },
+        ],
+        'abstracts': [
+            {
+                'value': 'We investigate the effects of (...)',
+            },
+        ],
+    }
+
+    expected = {
+        'title': 'Effects of top compositeness',
+        'abstract': 'We investigate the effects of (...)',
+    }
+    result = prepare_payload(record)
+
+    assert expected == result
+
+
+@patch('inspirehep.modules.workflows.tasks.classifier.get_classifier_url')
+def test_guess_coreness_fails_without_a_classifier_url(g_b_u):
+    g_b_u.return_value = ''
+
+    obj = MockObj({}, {})
+    eng = MockEng()
+
+    assert guess_coreness(obj, eng) is None
+    assert 'relevance_prediction' not in obj.extra_data
+
+
+@patch('inspirehep.modules.workflows.tasks.classifier.get_classifier_url')
+def test_guess_coreness_fails_when_request_fails(mocked_get_classifier_url):
+    mocked_get_classifier_url.return_value = 'https://inspire-classifier.web.cern.ch/api/predict/does-not-exist'
+
+    obj = MockObj({}, {})
+    eng = MockEng()
+
+    with pytest.raises(Exception):
+        assert guess_coreness(obj, eng) is None
```

Comment:
you should remove the `assert` and just call `guess_coreness` here, as you're in fact asserting that calling `guess_coreness` raises here, not that the assertion you write is correct. It's also better to use the more specific `requests.RequestException` instead of `Exception`.